File: iOS/PanoAudioChat/PanoAudioChatSDK/Source/Sections/AudioRoom/View/ApplyUserList/PanoApplyUserListCell.m
//
//  PanoApplyUserListCell.m
//  PanoAudioChatSDK
//
//  Created by pano on 2021/7/19.
//

#import "PanoApplyUserListCell.h"
#import "PanoDefine.h"
#import <Masonry/Masonry.h>



@implementation PanoApplyUserListCell {
    UIButton *acceptBtn;
    UIButton *rejectBtn;
    UIView *lineView;
}

- (instancetype)initWithStyle:(UITableViewCellStyle)style reuseIdentifier:(NSString *)reuseIdentifier {
    self = [super initWithStyle:style reuseIdentifier:reuseIdentifier];
    lineView = [UIView new];
    lineView.backgroundColor = defaultBorderColor();
    [self.contentView addSubview:lineView];
    
    nameLabel = [UILabel new];
    nameLabel.font = fontMedium();
    nameLabel.textColor = defaultTextColor();
    [self.contentView addSubview:nameLabel];
    
    acceptBtn = [UIButton buttonWithType:UIButtonTypeCustom];
    [acceptBtn setTitle:NSLocalizedString(@"允许", nil) forState: UIControlStateNormal];
    acceptBtn.titleLabel.textColor = [UIColor whiteColor];
    [acceptBtn setBackgroundImage:[UIImage imageNamed:@"home_bg_blue"] forState: UIControlStateNormal];
    [acceptBtn setBackgroundImage:[UIImage imageNamed:@"home_bg_blue"] forState: UIControlStateHighlighted];
    [acceptBtn addTarget:self action:@selector(accept:) forControlEvents:UIControlEventTouchUpInside];
    acceptBtn.layer.masksToBounds = true;
    acceptBtn.layer.cornerRadius = 15.0;
    [self.contentView addSubview:acceptBtn];
    
    rejectBtn = [UIButton buttonWithType:UIButtonTypeCustom];
    [rejectBtn setTitle:NSLocalizedString(@"拒绝", nil) forState: UIControlStateNormal];
    [rejectBtn setTitleColor:defaultTextColor() forState:UIControlStateNormal];
    [rejectBtn setTitleColor:defaultTextColor() forState:UIControlStateHighlighted];
    [rejectBtn addTarget:self action:@selector(reject:) forControlEvents:UIControlEventTouchUpInside];
    [self.contentView addSubview:rejectBtn];
    
    [lineView mas_makeConstraints:^(MASConstraintMaker *make) {
        make.left.bottom.right.mas_equalTo(0).insets(UIEdgeInsetsMake(0, 0, 0.8, 0));
        make.height.mas_equalTo(0.8);
    }];
    
    [nameLabel mas_makeConstraints:^(MASConstraintMaker *make) {
        make.centerY.mas_equalTo(0);
        make.left.mas_equalTo(0).offset(defaultMargin);
        make.right.mas_equalTo(rejectBtn.mas_left).offset(-defaultMargin);
    }];
    
    [acceptBtn mas_makeConstraints:^(MASConstraintMaker *make) {
        make.centerY.mas_equalTo(0);
        make.right.mas_equalTo(0).offset(-defaultMargin);
        make.height.mas_equalTo(30);
        make.width.mas_equalTo(84);
    }];
    
    [rejectBtn mas_makeConstraints:^(MASConstraintMaker *make) {
        make.centerY.mas_equalTo(0);
        make.right.mas_equalTo(acceptBtn.mas_left).offset(-defaultMargin);
        make.height.mas_equalTo(acceptBtn);
        make.width.mas_equalTo(acceptBtn);
    }];
    return self;
}

- (void)accept:(UIButton *)sender {
    acceptBlock();
}

- (void)reject:(UIButton *)sender {
    rejectBlock();
}

@end

File: iOS/PanoAudioChat/PanoAudioChat/Classes/Sections/AudioRoom/View/ApplyUserList/PanoApplyUserListCell.swift
//
//  ApplyUserListCell.swift
//  PanoAudioChat
//
//  Created by wenhua.zhang on 2020/11/3.
//

import UIKit


typealias PanoApplyButtonClosure = (() -> Void)
class PanoApplyUserListCell : UITableViewCell {

    var acceptBtn : UIButton!
    var rejectBtn : UIButton!
    var nameLabel : UILabel!
    var lineView : UIView!
    var acceptBlock : PanoApplyButtonClosure!
    var rejectBlock : PanoApplyButtonClosure!
    
    override init(style: UITableViewCell.CellStyle, reuseIdentifier: String?) {
        super.init(style: style, reuseIdentifier: reuseIdentifier);
        
        lineView = UIView()
        lineView.backgroundColor = DefaultBorderColor
        self.contentView.addSubview(lineView)
        
        nameLabel = UILabel()
        nameLabel.font = FontMedium
        nameLabel.textColor = DefaultTextColor
        self.contentView.addSubview(nameLabel)
        
        acceptBtn = UIButton()
        acceptBtn.setTitle(NSLocalizedString("允许", comment: ""), for: .normal)
        acceptBtn.titleLabel?.textColor = UIColor.white
        acceptBtn.setBackgroundImage(UIImage(named: "home_bg_blue"), for: .normal)
        acceptBtn.setBackgroundImage(UIImage(named: "home_bg_blue"), for: .highlighted)
        acceptBtn.addTarget(self, action: #selector(accept(sender:)), for: .touchUpInside)
        acceptBtn.layer.masksToBounds = true
        acceptBtn.layer.cornerRadius = 15.0
        self.contentView.addSubview(acceptBtn)
        
        rejectBtn = UIButton()
        rejectBtn.setTitle(NSLocalizedString("拒绝", comment: ""), for: .normal)
        rejectBtn.setTitleColor(DefaultTextColor, for: .normal)
        rejectBtn.setTitleColor(DefaultTextColor, for: .highlighted)
        rejectBtn.addTarget(self, action: #selector(reject(sender:)), for: .touchUpInside)
        self.contentView.addSubview(rejectBtn)
        
        lineView.snp.makeConstraints { (make) in
            make.left.bottom.right.equalToSuperview().inset(UIEdgeInsets(top: 0, left: 0, bottom: 0.8, right: 0))
            make.height.equalTo(0.8);
        }
        
        nameLabel.snp.makeConstraints { (make) in
            make.centerY.equalToSuperview()
            make.left.equalToSuperview().offset(defaultMargin)
            make.right.equalTo(rejectBtn.snp.left).offset(-defaultMargin)
        }
        
        acceptBtn.snp.makeConstraints { (make) in
            make.centerY.equalToSuperview()
            make.right.equalToSuperview().offset(-defaultMargin)
            make.height.equalTo(30)
            make.width.equalTo(84)
        }
        
        rejectBtn.snp.makeConstraints { (make) in
            make.centerY.equalToSuperview()
            make.right.equalTo(acceptBtn.snp.left).offset(-defaultMargin)
            make.height.equalTo(acceptBtn)
            make.width.equalTo(acceptBtn)
        }
    }
    
    required init?(coder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    @objc func accept(sender: UIButton!) {
        acceptBlock();
    }

    @objc func reject(sender: UIButton!) {
        rejectBlock();
    }
}
